common.js

```diff
@@ -49,3 +49,47 @@ function if_auto_decrypt(cb) {
         }
     });
 }
+
+// See https://gist.github.com/samgiles/762ee337dff48623e729
+// [B](f: (A) ⇒ [B]): [B]  ; Although the types in the arrays aren't strict (:
+Array.prototype.flatMap = function(lambda) {
+    return Array.prototype.concat.apply([], this.map(lambda));
+};
+
+function sanitize(node) {
+    return node.text().trim().replace(/\s/g, '');
+}
+
+function handle_response(textToDecrypt, response, prompt_on_fail, callback) {
+  switch (response.code) {
+    case "ok":
+      callback(response.text);
+      break;
+    case "bad_password":
+      if (prompt_on_fail) {
+        prompt_and_decrypt(textToDecrypt, callback);
+      }
+      break;
+    default:
+      if (prompt_on_fail) {
+        display_error(response.error);
+      }
+  }
+}
+
+function prompt_and_decrypt(textToDecrypt, callback) {
+  prompt_password(function(pw) {
+    do_decrypt(textToDecrypt, pw, true, callback);
+  });
+}
+
+function do_decrypt(textToDecrypt, password, prompt_on_fail, callback) {
+  chrome.runtime.sendMessage({
+    op: 'decrypt',
+    alg: 'VaultAES256',
+    data: textToDecrypt,
+    password: password
+  }, function(response) {
+    handle_response(textToDecrypt, response, prompt_on_fail, callback);
+  });
+}
```

Comment:
Main reason these were in `GH` was to try and minimize the number of names added to the page's namespace, and reduce the potential for conflicts there down the line.. it might be better to move everything in common into an AnsibleVaultDecryptor object to namespace it.